src/components/token/TokenBalances.js

```diff
@@ -0,0 +1,327 @@
+    const newURL = this.pagination.setURLParameters({
+      sortBy: this.state.sortBy,
+      order: this.state.order,
+    });
+
+    this.props.history.push(newURL);
+  }
+
+  /**
+    * Process events when next page is requested by user
+    */
+  nextPageClicked = async () => {
+    this.setState({ calculatingPage: true });
+
+    const nextPage = this.state.page + 1;
+    let searchAfter = get(find(this.state.pageSearchAfter, { page: nextPage }), 'searchAfter', []);
+
+    // Calculate searchAfter of next page if not already calculated
+    if (isEmpty(searchAfter)) {
+      const lastCurrentTokenSort = get(last(this.state.tokenBalances), 'sort', []);
+
+      const newEntry = {
+        page: nextPage,
+        searchAfter: lastCurrentTokenSort,
+      }
+
+      this.setState({
+        pageSearchAfter: [
+          ...this.state.pageSearchAfter,
+          newEntry,
+        ],
+      });
+
+      searchAfter = lastCurrentTokenSort;
+    }
+
+    const tokenBalances = await this.getTokenBalances(searchAfter);
+
+    this.setState({
+      tokenBalances: tokenBalances.hits,
+      hasAfter: tokenBalances.has_next,
+      hasBefore: true,
+      page: nextPage,
+      calculatingPage: false,
+    });
+  }
+
+  /**
+   * Process events when previous page is requested by user
+   */
+  previousPageClicked = async () => {
+    this.setState({ calculatingPage: true });
+
+    const previousPage = this.state.page - 1;
+    const searchAfter = get(find(this.state.pageSearchAfter, { page: previousPage }), 'searchAfter', []);
+    const tokenBalances = await this.getTokenBalances(searchAfter);
+
+    this.setState({
+      tokenBalances: tokenBalances.hits,
+      hasAfter: true,
+      hasBefore: previousPage === 1 ? false : true,
+      page: previousPage,
+      calculatingPage: false,
+    });
+  }
+
+  onTokenSelected = async (token) => {
+    if (!token) {
+      await this.setState({
+        tokenId: '00'
+      });
+
+      this.performSearch();
+      return;
+    }
+
+    await this.setState({
+      tokenId: token.id,
+    });
+
+    this.performSearch();
+  }
+
+  /**
+   * Process table header click. This indicates that user wants data to be sorted by a determined field
+   *
+   * @param {*} event
+   * @param {*} header
+   */
+  tableHeaderClicked = async (_event, header) => {
+    if (header === this.state.sortBy) {
+      await this.setState({ order: this.state.order === 'asc' ? 'desc' : 'asc' });
+    } else {
+      await this.setState({ sortBy: header, order: 'asc' });
+    }
+
+    await this.performSearch();
+  }
+
+  render() {
+    if (this.state.maintenanceMode) {
+      return <ErrorMessageWithIcon message='This feature is under maintenance. Please try again after some time' />;
+    }
+
+    const renderSearchField = () => {
+      return <TokenAutoCompleteField onTokenSelected={this.onTokenSelected.bind(this)}/>;
+    };
+
+    const renderTokensTable = () => {
+      if (this.state.maintenanceMode) {
```

Comment:
Why to we need this? Won't we already return the `<ErrorMessageWithIcon />` above in case this is enabled?

Reply:
You are right, fixed in 2bd55756c14c3e489b1f21238182fb2fe8c0a46c